tools/generate/src/generate-adaptor.ts

```diff
@@ -49,10 +49,21 @@ const copyAndRename = async (
       await fs.mkdir(targetPath, { recursive: true });
       await copyAndRename(sourcePath, targetPath, adaptorName);
     } else if (item.isFile()) {
-      let content = await fs.readFile(sourcePath, 'utf8');
-      content = content.replace(/{{TEMPLATE}}/g, adaptorName);
-
-      await fs.writeFile(targetPath, content);
+      const ext = path.extname(item.name).toLowerCase();
+      if (
+        ext === '.png' ||
+        ext === '.jpg' ||
+        ext === '.jpeg' ||
+        ext === '.gif' ||
+        ext === '.svg'
+      ) {
+        const content = await fs.readFile(sourcePath);
+        await fs.writeFile(targetPath, content);
+      } else {
+        let content = await fs.readFile(sourcePath, 'utf8');
+        content = content.replace(/{{TEMPLATE}}/g, adaptorName);
+        await fs.writeFile(targetPath, content);
```

Comment:
Because the `writeFile` line is the same, I would declare content as a `let` outside the if/else block, read and maybe transform the content INSIDE the blocks, and then write the file AFTER the if/else. So that you're not repeating any code :sunglasses: